frontend/src/stores/roles/actions.ts

```diff
@@ -23,3 +23,23 @@ function setRoles(newRoles: Role[]): void {
 }
 
 export { fetchRoles, addRole }
+
+export const saveKnowledge = async (newKnowledge: string) => {
+  try {
+    const response = await fetch('/api/knowledges', {
+      method: 'POST',
+      headers: {
+        'Content-Type': 'application/json'
+      },
+      body: JSON.stringify({ name: newKnowledge })
+    })
+    if (!response.ok) {
```

Comment:
Please, add error treatment here (check if the error is related to

1. a knowledge with the same name already exists
2. is it a technical error
3. etc), throwing the messages to the user when necessary (ElMessage)

Reply:
done